api/jobset/v1alpha2/jobset_types.go

```diff
@@ -458,6 +469,46 @@ type Coordinator struct {
 	PodIndex int `json:"podIndex,omitempty"`
 }
 
+// volumeClaimPolicy defines volume claim templates and lifecycle management for shared PVCs.
+type VolumeClaimPolicy struct {
+	// templates is a list of shared PVC claims that ReplicatedJobs are allowed to reference.
+	// The JobSet controller is responsible for creating shared PVCs that can be mounted by
+	// multiple ReplicatedJobs. Every claim in this list must have a matching (by name)
+	// volumeMount in one container or initContainer in at least one ReplicatedJob template.
+	// ReplicatedJob template must not have volumes with the same name as defined in this template.
+	// Generated PVC naming convention: <claim-name>-<jobset-name>
+	// Example: "model-cache-trainjob" (shared volume across all ReplicatedJobs).
+	// +kubebuilder:validation:MaxItems=50
+	// +listType=atomic
```

Comment:
I'm trying to fix the linter but I think it would complain here once I enabled Option or required settings.

Each field should have an optional tag.